### ISSUE TITLE
In MessageInput turning disableAttachments=true produces render errors

### ISSUE THREAD

**Issue by mingzew** [GetStream/stream-chat-flutter#668]
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

Android

### What version are you using?

2.2.1

### What happened?

I flipped MessageInput(disableAttachments: true), when the chat page loaded up, it produced errors mentioned below. I am using Pixel XL (mobile)

### Steps to reproduce

```
1. Create a 
StreamChannel(
                channel: _channel,
                child: Column(
                  children: [
                    chatAppBar(),
                    Expanded(child: MessageListView(footerBuilder: (context) {
                      return MessageInput(disableAttachments: true);
                    })),
                  ],
                ));
2. load up the page
3. Kaboom    
```

### Relevant log output

```shell
======== Exception caught by rendering library =====================================================
The following assertion was thrown during performLayout():
'package:flutter/src/rendering/box.dart': Failed assertion: line 320 pos 12: 'width > 0.0': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

The relevant error-causing widget was: 
  MessageInput file:///Users/admin/git/hype_dating_UI/lib/pages/chat.dart:271:30
When the exception was thrown, this was the stack: 
#2      BoxConstraints.constrainSizeAndAttemptToPreserveAspectRatio (package:flutter/src/rendering/box.dart:320:12)
#3      RenderFittedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:2529:30)
#4      RenderObject.layout (package:flutter/src/rendering/object.dart:1779:7)
#5      RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:116:14)
#6      RenderObject.layout (package:flutter/src/rendering/object.dart:1779:7)
...
The following RenderObject was being processed when the exception was fired: RenderFittedBox#156f9 relayoutBoundary=up20 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
...  parentData: <none> (can use size)
...  constraints: BoxConstraints(unconstrained)
...  size: MISSING
...  fit: contain
...  alignment: Alignment.center
...  textDirection: ltr
  child: RenderFlex#56109 relayoutBoundary=up21 NEEDS-PAINT
    parentData: <none> (can use size)
    constraints: BoxConstraints(unconstrained)
    size: Size(0.0, 0.0)
    direction: horizontal
    mainAxisAlignment: spaceEvenly
    mainAxisSize: max
    crossAxisAlignment: center
    textDirection: ltr
    verticalDirection: down
RenderObject: RenderFittedBox#156f9 relayoutBoundary=up20 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
  parentData: <none> (can use size)
  constraints: BoxConstraints(unconstrained)
  size: MISSING
  fit: contain
  alignment: Alignment.center
  textDirection: ltr
  child: RenderFlex#56109 relayoutBoundary=up21 NEEDS-PAINT
    parentData: <none> (can use size)
    constraints: BoxConstraints(unconstrained)
    size: Size(0.0, 0.0)
    direction: horizontal
    mainAxisAlignment: spaceEvenly
    mainAxisSize: max
    crossAxisAlignment: center
    textDirection: ltr
    verticalDirection: down
====================================================================================================

======== Exception caught by rendering library =====================================================
The following assertion was thrown during performLayout():
RenderBox was not laid out: RenderFittedBox#156f9 relayoutBoundary=up20 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
'package:flutter/src/rendering/box.dart':
Failed assertion: line 1930 pos 12: 'hasSize'


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

The relevant error-causing widget was: 
  MessageInput file:///Users/admin/git/hype_dating_UI/lib/pages/chat.dart:271:30
When the exception was thrown, this was the stack: 
#2      RenderBox.size (package:flutter/src/rendering/box.dart:1930:12)
#3      RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:117:21)
#4      RenderObject.layout (package:flutter/src/rendering/object.dart:1779:7)
#5      RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:116:14)
#6      RenderObject.layout (package:flutter/src/rendering/object.dart:1779:7)
...
The following RenderObject was being processed when the exception was fired: RenderAnimatedOpacity#fed10 relayoutBoundary=up19 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
...  parentData: <none> (can use size)
...  constraints: BoxConstraints(unconstrained)
...  size: MISSING
...  opacity: AnimationController#c74f7(⏭ 1.000; paused)➩CurveTween(curve: _Linear)➩1.0
RenderObject: RenderAnimatedOpacity#fed10 relayoutBoundary=up19 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
  parentData: <none> (can use size)
  constraints: BoxConstraints(unconstrained)
  size: MISSING
  opacity: AnimationController#c74f7(⏭ 1.000; paused)➩CurveTween(curve: _Linear)➩1.0
...  child: RenderFittedBox#156f9 relayoutBoundary=up20 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
...    parentData: <none> (can use size)
...    constraints: BoxConstraints(unconstrained)
...    size: MISSING
...    fit: contain
...    alignment: Alignment.center
...    textDirection: ltr
...    child: RenderFlex#56109 relayoutBoundary=up21 NEEDS-PAINT
...      parentData: <none> (can use size)
...      constraints: BoxConstraints(unconstrained)
...      size: Size(0.0, 0.0)
...      direction: horizontal
...      mainAxisAlignment: spaceEvenly
...      mainAxisSize: max
...      crossAxisAlignment: center
...      textDirection: ltr
...      verticalDirection: down
====================================================================================================
```

### Flutter doctor output

```shell
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-x64, locale en-US)
    • Flutter version 2.2.3 at /Users/admin/Development/flutter
    • Framework revision f4abaa0735 (10 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/admin/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.40.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Pixel XL (mobile)                     • HT68S0203969  • android-arm64  • Android 10 (API 29)
    • Android SDK built for x86 64 (mobile) • emulator-5554 • android-x64    • Android 10 (API 29) (emulator)
    • Chrome (web)                          • chrome        • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by xsahil03x** [comment]
Hey @mingzew, not able to reproduce the issue. Working fine for me.
Can you share a small project to test the issue?

**Comment by mingzew** [comment]
Hi @xsahil03x thanks for the follow up. I did some testing and narrowed down the issue. These are the steps to create this issue:

1. Create a new channel type on web console
2. After creation, go into the channel and switch commands to "off"
3. Make sure your ```MessageInput(disableAttachments: true)```

I've attached a sample project, just replace APIKEY and CHANNEL_TYPE in stream_api.dart

[my_app.zip](https://github.com/GetStream/stream-chat-flutter/files/7140098/my_app.zip)
